pkg/backup/resource_backupper.go

```diff
@@ -243,37 +246,85 @@ func (rb *defaultResourceBackupper) backupResource(group *metav1.APIResourceList
+	// one item and IncludeClusterResources is nil. If IncludeClusterResources is false
+	// we don't want to back it up, and if it's true it will already be included.
+	if backedUpItem && rb.backupRequest.Spec.IncludeClusterResources == nil {
+		rb.backupCRD(log, gr, itemBackupper)
+	}
 
-			err = itemBackupper.backupItem(log, unstructured, gr)
-			if aggregate, ok := err.(kubeerrs.Aggregate); ok {
-				log.WithField("name", metadata.GetName()).Infof("%d errors encountered backup up item", len(aggregate.Errors()))
-				// log each error separately so we get error location info in the log, and an
-				// accurate count of errors
-				for _, err = range aggregate.Errors() {
-					log.WithError(err).WithField("name", metadata.GetName()).Error("Error backing up item")
-				}
+	return nil
+}
 
-				continue
-			}
-			if err != nil {
-				log.WithError(err).WithField("name", metadata.GetName()).Error("Error backing up item")
-				continue
-			}
+func (rb *defaultResourceBackupper) backupItem(
+	log logrus.FieldLogger,
+	gr schema.GroupResource,
+	itemBackupper ItemBackupper,
+	unstructured runtime.Unstructured,
+) bool {
+	metadata, err := meta.Accessor(unstructured)
+	if err != nil {
+		log.WithError(errors.WithStack(err)).Error("Error getting a metadata accessor")
+		return false
+	}
+
+	if gr == kuberesource.Namespaces && !rb.backupRequest.NamespaceIncludesExcludes.ShouldInclude(metadata.GetName()) {
+		log.WithField("name", metadata.GetName()).Info("Skipping namespace because it's excluded")
+		return false
+	}
+
+	backedUpItem, err := itemBackupper.backupItem(log, unstructured, gr)
+	if aggregate, ok := err.(kubeerrs.Aggregate); ok {
+		log.WithField("name", metadata.GetName()).Infof("%d errors encountered backup up item", len(aggregate.Errors()))
+		// log each error separately so we get error location info in the log, and an
+		// accurate count of errors
+		for _, err = range aggregate.Errors() {
+			log.WithError(err).WithField("name", metadata.GetName()).Error("Error backing up item")
 		}
+
+		return false
 	}
+	if err != nil {
+		log.WithError(err).WithField("name", metadata.GetName()).Error("Error backing up item")
+		return false
+	}
+	return backedUpItem
+}
 
-	return nil
+// Adds CRD to the backup if one is found corresponding to this resource
+func (rb *defaultResourceBackupper) backupCRD(
+	log logrus.FieldLogger,
+	gr schema.GroupResource,
+	itemBackupper ItemBackupper,
+) {
+	crdGr := schema.GroupResource{Group: apiextv1beta1.GroupName, Resource: "customresourcedefinitions"}
+	crdClient, err := rb.dynamicFactory.ClientForGroupVersionResource(apiextv1beta1.SchemeGroupVersion,
+		metav1.APIResource{
+			Name:       "customresourcedefinitions",
+			Namespaced: false,
+		},
+		"",
+	)
+	if err != nil {
+		log.WithError(err).Error("Error getting dynamic client for CRDs")
+		return
+	}
+
+	unstructured, err := crdClient.Get(gr.String(), metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
```

Comment:
Why do we not log if the CRD was not found?

Reply:
Answered my own question: This means the resource isn't a CR. So extra logging would be noisy and pointless